javascript/widgets/handler/callback.js

```diff
@@ -92,6 +92,19 @@ firebaseui.auth.widget.handler.handleCallback =
         firebaseui.auth.widget.handler.handleCallbackFailure_(
             app, component, /** @type {!Error} */ (error));
       }
+    } else if (error &&
+        error['code'] == 'auth/operation-not-supported-in-this-environment' &&
+        firebaseui.auth.widget.handler.common.isPasswordProviderOnly(app)) {
```

Comment:
Instead of actually calling getRedirectResult, it may be better to return quickly when isPasswordProviderOnly is true.

You could do that before the promise resolution:

```
if (!opt_result && firebaseui.auth.widget.handler.common.isPasswordProviderOnly(app)) {
  firebaseui.auth.widget.handler.handleCallbackResult_(
      app,
      component,
       {
         'user': null
       });
  return;
}
```

Not even sure the `!opt_result` condition is necessary.